### PR TITLE
Updated gophercloud to lastest master

### LINKED ISSUE #848: Update Gophercloud
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind feature

**What happened**:
A change was introduced in Gophercloud that could solve some problems we are facing while using openstack-cloud-provider.
**What you expected to happen**:
Gophercloud dependency gets updated to its current master.

**Anything else we need to know?**:
This change (https://github.com/gophercloud/gophercloud/issues/1764) would solve some problems we have in our environment. We would like openstack-cloud-provider to have those fixes.
**Environment**:
- openstack-cloud-controller-manager version: 1.16
- OS (e.g. from /etc/os-release): SLES 15.1
- Kernel (e.g. `uname -a`): 4.19

### PR DESCRIPTION
**Pull request by chrigl**
**What this PR does / why we need it**:

Get latest gophercloud fixes

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #848 

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -226,6 +226,8 @@ github.com/gophercloud/gophercloud v0.0.0-20190212181753-892256c46858/go.mod h1:
 github.com/gophercloud/gophercloud v0.1.0/go.mod h1:vxM41WHh5uqHVBMZHzuwNOHh8XEoIEcSTewFxm1c5g8=
 github.com/gophercloud/gophercloud v0.6.1-0.20191025185032-6ad562af8c1f h1:oUSh4Z++AIcYE3FClYjEqC+T/esZ2ZDaDeaEeXQ2mkE=
```
Comment (ramineni) [inline go.sum]: old entries of gophercloud are still required?

Comment (kayrus) [reply go.sum]: this is only a checksum

Comment (chrigl) [reply go.sum]: Ran go mod tidy

Comment (ramineni) [reply go.sum]: +1, Thanks

**Pull request merged**